src/internal/config/config.go

```diff
@@ -270,20 +269,24 @@ func (c *Config) write() error {
 	if err = tmpfile.Close(); err != nil {
 		return errors.EnsureStack(err)
 	}
-	if err = os.Rename(tmpfile.Name(), p); err != nil {
+	if err = os.Rename(tmpfile.Name(), path); err != nil {
 		// A rename could fail if the temporary directory is mounted on a
 		// different device than the config path. If the rename failed, try to
 		// just copy the bytes instead. Note that a destructive disk error could
 		// leave cachedConfig out of date.
 		// TODO(msteffen) attempt to backup the config if it exists & restore on
 		// failure.
-		if err = ioutil.WriteFile(p, rawConfig, 0644); err != nil {
-			return errors.Wrapf(err, "failed to copy updated config file from %s to %s", tmpfile.Name(), p)
+		if err = ioutil.WriteFile(path, rawConfig, 0644); err != nil {
+			return errors.Wrapf(err, "failed to copy updated config file from %s to %s", tmpfile.Name(), path)
 		}
 	}
 
 	// essentially short-cuts reading the new config back from disk
-	cachedConfig = proto.Clone(c).(*Config)
+	// Note: this will only fail in the case that the caller passes
```

Comment:
Just to clarify, the failure is due to when the caller passes in some random path that is not consistent with `configPath()`? Should we error in this case?

Reply:
maybe fail is the wrong word in the comment. The intention behind these changes is to expand the PachConfig API so that we can modify configs at arbitrary paths from go code. The benefit here is for testing. Basically we don't care to cache the config globally in the case that we are passing an arbitrary pach_config path from the API

Reply:
maybe fail is the wrong word in the comment. The intention behind these changes is to expand the PachConfig API so that we can modify configs at arbitrary paths from go code. The benefit here is for testing. Basically we don't care to cache the config globally in the case that we are passing an arbitrary pach_config path from the API